board/arm-64bit-toolchain.cmake

```diff
@@ -0,0 +1,59 @@
+## This file handles cross-compilation configurations for aarch64,
+## known as arm64. The objective of this file is to find and assign
+## cross-compiler and the entire toolchain.
+## It works best with buildroot toolchain, when using it the user
+## needs to set the: TOOLCHAIN_PREFIX and CMAKE_SYSROOT from the
+## command line.
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR ARM)
+
+if(MINGW OR CYGWIN OR WIN32)
+    set(UTIL_SEARCH_CMD where)
+elseif(UNIX OR APPLE)
+    set(UTIL_SEARCH_CMD which)
+endif()
+
+if(DEFINED TOOLCHAIN_PREFIX AND DEFINED CMAKE_SYSROOT)
+  message(STATUS "TOOLCHAIN_PREFIX is defined as: ${TOOLCHAIN_PREFIX}")
+  message(STATUS "CMAKE_SYSROOT is defined as: ${CMAKE_SYSROOT}")
+else ()
```

Comment:
Do you think we should issue an error if only one of `TOOLCHAIN_PREFIX` or `CMAKE_SYSROOT` is set?  I had some confusing results when I was setting `TOOLCHAIN_PREFIX` but not `CMAKE_SYSROOT` (and vice versa).  In that situation, it seems the setting of `TOOLCHAIN_PREFIX` and `CMAKE_SYSROOT` get overwritten.

Maybe it could be better to have one block for `if (DEFINED TOOLCHAIN_PREFIX)` and one for `if (DEFINED CMAKE_SYSROOT)`?

Reply:
Yes, we need to issue and error and stop cmake from continuing if one of them is set. Otherwise, this might lead to a very strange result.